### PR TITLE
Add support for Permanent Delegate extension

### PR DESCRIPTION
**Pull request by Aiyualive**
Changes
- Add additional_extensions to CreateMintArgs
- Add optional permanent_delegate account
- Add initialization for permanent_delegate extension. If no perm delegate was provided return an error

Assumptions:
- `update_account_lamports_to_minimum_balance` will add extra sizing for added extensions

Blockers:
- Anchor cannot serialize CreateMintArgs with `additional_extensions: ExtensionsVec`, since `ExtensionType` is not supported by Anchor

### CODE REVIEW

**Comment by Bhargavamacha** [comment]
Hey @Aiyualive thanks for the PR! Could you checkout [this](https://github.com/bridgesplit/anchor/blob/master/tests/spl/token-extensions/programs/token-extensions/src/instructions.rs) reference implementation and try to initialize permanent delegate instead of executing a separation instruction. Let me know if you have any issues.

**Comment by Aiyualive** [comment]
Gm @Bhargavamacha! Thank you for swiftly looking into this!

I just looked into your reference and have the following questions:
- All mints are forced to have those specified MINT_EXTENSIONS, which is not that flexible. Wouldn't it be better to allow for some more flexibility in which extensions you would want?

- Since permanent delegate can never be changed, I think it would be nice to have this account be passed in as an optional account. What do you think?

- If the permanent_delegate is optional, how would you pass that into the macro:
 extensions::permanent_delegate::delegate = permanent_delegate.key()?

**Comment by luke-truitt** [comment]
Bridgesplit is going to update the Anchor fork to support optional extension initialization in the macro, once this is live it should resolve this issue. Will tag here once that's complete

**Comment by Aiyualive** [comment]
- I wanted to assign `Pubkey::default()` as the default permanent delegate in case no argument was passed in, although this led to an invalid program argument. We experienced similar issues previously when trying ti simply assign the default pubkey as the transfer hook program. So I think there are some limitations with the use thereof.

- I also added accommodated some tests for the manager account for local testing.

File: programs/wen_new_standard/src/instructions/mint/create.rs [hunk 1]
```diff
@@ -49,6 +50,7 @@ pub struct CreateMintAccount<'info> {
         extensions::metadata_pointer::metadata_address = mint,
         extensions::group_member_pointer::authority = manager,
         extensions::transfer_hook::authority = authority,
+        extensions::permanent_delegate::delegate = args.permanent_delegate.unwrap_or_else(|| manager.key()),
```
Comment (Bhargavamacha) [inline programs/wen_new_standard/src/instructions/mint/create.rs:53]: I dont think we should leave permanent delegate as manager if user chose not to use it, can you add an update permanent delegate below to remove it if the argument passed is None

File: clients/wns-sdk/tests/e2e.test.ts [hunk 2]
```diff
@@ -56,7 +79,7 @@ describe('e2e tests', () => {
 		const messageV0 = new TransactionMessage({
 			payerKey: setup.payer.publicKey,
 			recentBlockhash: blockhash,
-			instructions: [createManagerIx, createGroupIx, addDistributionIx],
+			instructions: [createGroupIx, addDistributionIx],
```
Comment (Bhargavamacha) [inline clients/wns-sdk/tests/e2e.test.ts:82]: please run lint

**Review by Bhargavamacha — CHANGES_REQUESTED** [review]

**Review by luke-truitt — APPROVED** [review]

**Review by balmy-gazebo — APPROVED** [review]

**Pull request merged**